File: java/repository/src/main/java/validators/database/RallyDatabaseValidator.java
package validators.database;

import models.Rally;
import validators.interfaces.RallyValidator;

public class RallyDatabaseValidator implements RallyValidator {
    @Override
    public void validate(Rally rally) {
        if (rally.getEngineCapacity() <= 0)
            throw new InvalidModelException("Engine capacity can't be negative!");
    }
}

File: csharp/Repository/Validators/Database/RallyDatabaseValidator.cs
using Domain.Models;
using Repository.Validators.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Repository.Validators.Database.Postgresql
{
    public class RallyDatabaseValidator : IRallyValidator
    {
        public void Validate(Rally model)
        {
            if (model.EngineCapacity <= 0)
                throw new IRallyValidator.InvalidModelException("Engine capacity can't be negative!");
        }
    }
}
